src/app/reporting/EventLoggingTypes.h

```diff
@@ -0,0 +1,273 @@
+     * the power budget.
+     */
+    Info = 2,
+
+    /**
+     *  Debug priority denotes log entries of interest to the
+     *  developers of the system and is used primarily in the
+     *  development phase. Debug priority logs are
+     *  not accounted for in the bandwidth or power budgets of the
+     *  constrained devices; as a result, they must be used only over
+     *  a limited time span in production systems.
+     */
+    Debug                  = 3,
+    kPriorityLevel_Last    = Debug,
+    kPriorityLevel_Invalid = 4,
+
+} PriorityLevel;
+
+/**
+ * @brief
+ *   The structure that defines a schema for event metadata.
+ */
+struct EventSchema
+{
+    chip::ClusterId mClusterId; ///< ID of cluster
+    uint32_t mStructureType;    ///< Type of structure
+    PriorityLevel mPriority;    ///< Priority
+};
+
+/**
+ * @typedef timestamp_t
+ * Type used to describe the timestamp in milliseconds.
+ */
+typedef uint32_t timestamp_t;
+
+/**
+ * @typedef duration_t
+ * Type used to describe the duration, in milliseconds.
+ */
+typedef uint32_t duration_t;
+
+/**
+ * @typedef utc_timestamp_t
+ * Type used to describe the UTC timestamp in milliseconds.
+ */
+typedef uint64_t utc_timestamp_t;
+
+enum
+{
+    kTag_Region  = 1, ///< An 32-bit unsigned indicating the log region, i.e. the module to which the log message pertains.
+    kTag_Message = 2  ///< A string containing the actual debug message.
+};
+
+// Structures used to describe in detail additional options for event encoding
+
+/**
+ * @brief
+ *   The structure that provides a full resolution of the cluster.
+ */
+struct DetailedRootSection
+{
+    /**
+     * Default constructor
+     */
+    DetailedRootSection(void){};
+    chip::NodeId mNodeId;
+    chip::EndpointId mEndpointId;
+};
+
+/**
+ * @brief
+ *   The validity and type of timestamp included in EventOptions.
+ */
+typedef enum TimestampType
+{
+    kTimestampType_Invalid = 0,
+    kTimestampType_System,
+    kTimestampType_UTC
+} TimestampType;
+
+/**
+ * @brief
+ *   The union that provides an application set system or UTC timestamp.
+ */
+union Timestamp
+{
+    /**
+     * Default constructor.
+     */
+    Timestamp(void) : systemTimestamp(0){};
+
+    /**
+     * UTC timestamp constructor.
+     */
+    Timestamp(utc_timestamp_t aUtc) : utcTimestamp(aUtc){};
+
+    /**
+     * System timestamp constructor.
+     */
+    Timestamp(timestamp_t aSystem) : systemTimestamp(aSystem){};
+
+    timestamp_t systemTimestamp;  ///< System timestamp.
+    utc_timestamp_t utcTimestamp; ///< UTC timestamp.
+};
+
+/**
+ *   The structure that provides options for the different event fields.
+ */
+class EventOptions
+{
+public:
+    EventOptions(void);
+    EventOptions(bool);
+    EventOptions(timestamp_t);
+    EventOptions(utc_timestamp_t);
+    EventOptions(timestamp_t, bool);
+    EventOptions(utc_timestamp_t, bool);
+
+    Timestamp timestamp; /**< A union holding either system or UTC timestamp. */
```

Comment:
union is implementation detail. I would remove this comment.

Reply:
Done, thanks